packages/devextreme/js/__internal/ui/chat/messagelist.ts

```diff
@@ -169,6 +176,51 @@ class MessageList extends Widget<Properties> {
     });
   }
 
+  _shouldAddDateHeader(timestamp: undefined | string | number | Date): boolean {
+    const { showDateHeaders } = this.option();
+    if (timestamp === undefined || !showDateHeaders) {
+      return false;
+    }
+
+    const deserializedDate = dateSerialization.deserializeDate(timestamp);
+
+    if (isNaN(deserializedDate.getTime())) {
+      return false;
+    }
+
+    return !dateUtils.sameDate(this._messageDate, deserializedDate);
+  }
+
+  _createMessageDateHeader(timestamp: string | number | Date | undefined): void {
```

Comment:
Actually it's not related to the message directly. And i think it's better to use "day" everywhere because we also have "message datetime" and i'm worried naming will be very similar if we use "date"

```suggestion
  _createDayHeader(timestamp: string | number | Date | undefined): void {
```

Reply:
renamed everywhere